src/lib/mina_base/parties.ml

```diff
@@ -274,7 +274,17 @@ let check_depths (t : t) =
     true
   with _ -> false
 
-let check (t : t) = check_depths t
+(* Fee payer MUST use a full commitment, and other parties MAY use a full 
+   commitment IFF they are also the fee payer. *)
```

Comment:
Why is this restriction here? Is there some reason we can't allow other parties to opt into this?